api/desecapi/serializers.py

```diff
@@ -184,13 +268,24 @@ def to_internal_value(self, data):
         ret = []
         errors = []
 
-        partial = self.partial
+        known_rrsets = [(x.subname, x.type) for x in self.instance] if isinstance(self.instance, list) else []
+        known_rrsets_index = {(x.subname, x.type): x for x in self.instance} if isinstance(self.instance, list) else {}
```

Comment:
We have `known_rrsets == set(known_rrsets_index.keys())`, and `known_rrsets` is only used in one location below (in the assignment of `self.partial`. This is redundant and complicated; let's remove `known_rrsets`.

Reply:
Just saw that this is fixed in a later commit.